### PR TITLE
Level 2 interfaces

### PR DESCRIPTION
**Pull request by PWMaestro**
Created Vechile & Printer class.
Completed tasks accordint to level 2.

### CODE REVIEW

File: Autopark/Program.cs [hunk 1]
```diff
@@ -6,35 +6,52 @@ class Program
...
-                if (i == vechiles.Length - 1)
+                if (minMileage > vehicles[i].Mileage)
                 {
-                    vechiles[i].TaxCoefficient = 1.3;
+                    minMileage = vehicles[i].Mileage;
+                    indexOfMinMileage = i;
                 }
-                sumTaxCoefficient += vechiles[i].TaxCoefficient;
-                vechiles[i].Display();
             }
```
Comment (PushMeNow) [inline Autopark/Program.cs:49]: Move whole logic with searching and printing of max/min mileage in separate method

File: Autopark/Program.cs [hunk 2]
```diff
@@ -6,35 +6,52 @@ class Program
...
-            for (int i = 0; i < vechiles.Length; i++)
+            Vehicle[] vehicles = new Vehicle[] {
+                new Vehicle(vehicleTypes[0],"VW Crafter", "5427 AX-7", 2022, 2015, 376000, Color.Blue),
+                new Vehicle(vehicleTypes[0],"VW Crafter", "6427 AA-7", 2500, 2014, 227010, Color.White),
+                new Vehicle(vehicleTypes[0],"Electric Bus E321", "6785 BA-7", 12080, 2019, 20451, Color.Green),
+                new Vehicle(vehicleTypes[1],"Golf 5", "8682 AX-7", 1200, 2006, 230451, Color.Gray),
+                new Vehicle(vehicleTypes[1],"Tesla Model S 70 D", "E001 AA-7", 2200, 2019, 10454, Color.White),
+                new Vehicle(vehicleTypes[2],"Hamm HD 12 VV", null, 3000, 2016, 122, Color.Yellow),
+                new Vehicle(vehicleTypes[2],"МТЗ Беларус-1025.4", "1145 AB-7", 1200, 2020, 109, Color.Red)
+            };
```
Comment (PushMeNow) [inline Autopark/Program.cs:24]: Move it in static readonly field.

File: Autopark/Program.cs [hunk 3]
```diff
@@ -6,35 +6,52 @@ class Program
...
-                new VechileType("Bus", 1.2),
-                new VechileType("Car", 1.0),
-                new VechileType("Rink", 1.5),
-                new VechileType("Tractor", 1.2)
+            VehicleType[] vehicleTypes = new VehicleType[] {
+                new VehicleType("Bus", 1.2),
+                new VehicleType("Car", 1.0),
+                new VehicleType("Rink", 1.5),
+                new VehicleType("Tractor", 1.2)
             };
```
Comment (PushMeNow) [inline Autopark/Program.cs:14]: Move it in static readonly field

File: Autopark/Vehicle.cs [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        Red,
+        Orange,
+        Yellow,
+        Green,
+        Cyan,
+        Blue,
+        Violet,
+        Gray,
+        White
+    }
```
Comment (PushMeNow) [inline Autopark/Vehicle.cs:21]: Move it in separate file

File: Autopark/Vehicle.cs [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+    public class Vehicle : IComparable<Vehicle>
+    {
+        public readonly VehicleType type;
+        public readonly int manufactureYear;
+        public readonly string modelName;
```
Comment (PushMeNow) [inline Autopark/Vehicle.cs:27]: User mustn't know about these fields. Also could you add to these fields '_' prefix?

File: Autopark/Vehicle.cs [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+            double mileage,
+            Color color) 
+        {
+            this.type = type;
+            this.modelName = modelName;
+            this.RegistrationNumber = registrationNumber;
+            this.Weight = weight;
+            this.manufactureYear = manufactureYear;
+            this.Mileage = mileage;
+            this.Color = color;
```
Comment (PushMeNow) [inline Autopark/Vehicle.cs:53]: I don't see the reason in 'this' keyword here.

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi!
I have added some comments. Let me know if any questions.

**Review by dzmitri-litvinov — APPROVED** [review]

**Review by PushMeNow — APPROVED** [review]

**Pull request merged**